File: pi.c
#include <stdio.h> 

double leibniz_pi(long long n){
    double pi = 0.0;
    int sign = 1;

    for(long long i=0; i<n; i++){
        pi += sign * 1.0 / (2*i + 1);
        sign = -sign;
    }
    pi *= 4;
    return pi;
}

int main(){
    printf("%.30f\n", leibniz_pi(20000000000));
}

File: pi.go
package main

import "fmt"


func leibnizPi(n int) float64 {
    pi := 0.0
    sign := 1.0
    for i := 0; i < n; i++ {
        pi += sign / float64(2*i + 1)
        sign = -sign
    }
    return pi * 4
}
func main() {
    fmt.Println(leibnizPi(2000000000))
}